frontend/app-development/features/processEditor/ProcessEditor.tsx

```diff
@@ -60,6 +61,11 @@ export const ProcessEditor = (): React.ReactElement => {
     false,
   );
   const { data: layoutSets } = useLayoutSetsQuery(org, app);
+  const layoutSetsRef = React.useRef<LayoutSets>(layoutSets);
+
+  React.useEffect(() => {
+    layoutSetsRef.current = layoutSets;
+  }, [layoutSets]);
```

Comment:
Endrer dette noe som helst? Det vi sier her er bare "oppdater `layoutSetsRef.current` når komponenter laster på nytt med en ny verdi i `layoutSets`". Sånn sett bør det ikke være noen forskjell på å bruke `layoutSets` kontra `layoutSetsRef.current` lengre ned i komponenten.

Reply:
Nei, fordi jeg også bruker refen der instansen opprettes. Dersom man kun setter `layoutSets` direkte lenger ned så vil instansen aldri benytte den nye layoutSets-verdien fordi instansen kun opprettes første gang prosess-editoren rendres. 

Men det kan hende du har rett i at det ikke er nødvendig med en useEffect. Mener bare det var den eneste måten jeg fikk løst problemet 🤷

Reply:
Jeg ser bare at `layoutSetsRef` brukes der hvor `layoutSets` ble brukt tidligere. Og verdien i `layoutSetsRef` blir satt av nettopp `layoutSets`, både når `ref`en blir initialisert og når komponenten oppdaterer seg. Derfor har jeg vanskeligheter med å forstå hvordan dette endrer noe. Hvordan har du kommet frem til denne løsningen? Kan du lage en enhetstest som verifiserer at saken er løst?

Reply:
Jeg mener at det fungerte fordi instansen som sendes til processEditor ikke nå holder på layoutSets objektet, men heller på en referanse til det. Da den kun hadde layoutSets ble objektet i instansen `onProcessTaskRemove` aldri oppdatert. Men jeg kan prøve å lage en test. Tror dog det er vanskelig ettersom jeg er avhengig av å simulere at layoutSets blir endret fra utsiden. Mulig det lar seg gjøre med en rerender. 
Har du forsøkt å teste usecaset som er beskrevet i isseuet? Altså å legge til et datasteg og fjerne det. I staging vil man se at layoutsettet som ble lagt til av å legge til steget ikke ble fjernet når man navigerer man til utforming, feks.

Reply:
Slik jeg forstår koden, inneholder både `layoutSets` og `layoutSetsRef.current` den samme referansen, og så lenge det ikke er noe annet enn denne `useEffect`-hooken som har noen innvirkning på `layoutSets`, så bør disse to variablene være helt like. En ting som faktisk endrer seg her er at `ProcessEditorImpl` nå blir lastet med den _gamle_ referansen til `layoutSets`, siden funksjonen i `useEffect` blir kjørt etter lastingen, og det er først da `layoutSetsRef.current` blir oppdatert.

Nei, jeg har ikke testet, men jeg kan prøve, så kan jeg se om jeg også får til å lage en test. Det høres ut som at `rerender` er veien å gå. Hvis ingenting annet går med Jest, bør vi prøve med Playwright.